sonic-sensormond/scripts/sensormond

```diff
@@ -0,0 +1,548 @@
+            return False
+
+        self.under_threshold = status
+        return True
+
+
+#
+# voltageUpdater  ======================================================================
+#
+class VoltageUpdater(logger.Logger):
+    # Voltage information table name in database
+    VOLTAGE_INFO_TABLE_NAME = 'VOLTAGE_INFO'
+
+    def __init__(self, chassis):
+        '''
+        Initializer of VoltageUpdater
+        :param chassis: Object representing a platform chassis
+        '''
+        super(VoltageUpdater, self).__init__(SYSLOG_IDENTIFIER)
+
+        self.chassis = chassis
+        self.voltage_status_dict = {}
+        state_db = daemon_base.db_connect("STATE_DB")
+        self.table = swsscommon.Table(state_db, self.VOLTAGE_INFO_TABLE_NAME)
+        self.phy_entity_table = swsscommon.Table(state_db, PHYSICAL_ENTITY_INFO_TABLE)
+        self.chassis_table = None
+
+        self.is_chassis_system = chassis.is_modular_chassis()
+        if self.is_chassis_system:
+            self.module_voltage_sensors = set()
+            my_slot = try_get(chassis.get_my_slot, INVALID_SLOT)
+            if my_slot != INVALID_SLOT:
+                try:
+                    # Modular chassis may not have table CHASSIS_STATE_DB.
+                    table_name = VoltageUpdater.VOLTAGE_INFO_TABLE_NAME+'_'+str(my_slot)
+                    chassis_state_db = daemon_base.db_connect("CHASSIS_STATE_DB")
+                    self.chassis_table = swsscommon.Table(chassis_state_db, table_name)
+                except Exception as e:
+                    self.chassis_table = None
+
+    def __del__(self):
+        if self.table:
+            table_keys = self.table.getKeys()
+            for tk in table_keys:
+                self.table._del(tk)
+                if self.is_chassis_system and self.chassis_table is not None:
+                    self.chassis_table._del(tk)
+        if self.phy_entity_table:
+            phy_entity_keys = self.phy_entity_table.getKeys()
+            for pek in phy_entity_keys:
+                self.phy_entity_table._del(pek)
+
+    def _log_on_status_changed(self, normal_status, normal_log, abnormal_log):
+        '''
+        Log when any status changed
+        :param normal_status: Expected status.
+        :param normal_log: Log string for expected status.
+        :param abnormal_log: Log string for unexpected status
+        :return:
+        '''
+        if normal_status:
+            self.log_notice(normal_log)
+        else:
+            self.log_warning(abnormal_log)
+
+    def update(self):
+        '''
+        Update all voltage information to database
+        :return:
+        '''
+        self.log_debug("Start voltage update")
+        for index, voltage_sensor in enumerate(self.chassis.get_all_voltage_sensors()):
+
+            self._refresh_voltage_status(CHASSIS_INFO_KEY, voltage_sensor, index)
+
+        if self.is_chassis_system:
+            available_voltage_sensors = set()
```

Comment:
Why only handle this for chassis system? For NON chassis system, PSU also can be hot swapable.